Minio/Helper/Utils.cs

```diff
@@ -196,6 +197,20 @@ internal static bool IsSupersetOf(IList<string> l1, IList<string> l2)
         return !l2.Except(l1).Any();
     }
 
+    public static Task ForEachAsync<T>(this IEnumerable<T> source, int dop, Func<T, Task> body)
+    {
+        return Task.WhenAll(
```

Comment:
Add a `#IF NET6` here and use my method when NET6 or higher is available. It is much more efficient.